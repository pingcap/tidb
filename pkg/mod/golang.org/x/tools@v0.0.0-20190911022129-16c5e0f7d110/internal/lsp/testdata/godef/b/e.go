package b

import (
	"fmt"

	"golang.org/x/tools/internal/lsp/godef/a"
)

func useThings() {
	t := a.Thing{}      //@mark(bStructType, "ing")
	fmt.Print(t.Member) //@mark(bMember, "ember")
	fmt.Print(a.Other)  //@mark(bVar, "ther")
	a.Things()          //@mark(bFunc, "ings")
}

/*@
godef(bStructType, Thing)
godef(bMember, Member)
godef(bVar, Other)
godef(bFunc, Things)
*/
