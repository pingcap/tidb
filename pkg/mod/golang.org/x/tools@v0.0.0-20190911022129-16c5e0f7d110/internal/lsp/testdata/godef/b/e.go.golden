-- Member-definition --
godef/a/d.go:6:2-8: defined here as field Member string

-- Member-definition-json --
{
	"span": {
		"uri": "file://godef/a/d.go",
		"start": {
			"line": 6,
			"column": 2,
			"offset": 55
		},
		"end": {
			"line": 6,
			"column": 8,
			"offset": 61
		}
	},
	"description": "field Member string"
}

-- Member-hover --
@Member
field Member string
-- Other-definition --
godef/a/d.go:9:5-10: defined here as var a.Other a.Thing

-- Other-definition-json --
{
	"span": {
		"uri": "file://godef/a/d.go",
		"start": {
			"line": 9,
			"column": 5,
			"offset": 86
		},
		"end": {
			"line": 9,
			"column": 10,
			"offset": 91
		}
	},
	"description": "var a.Other a.Thing"
}

-- Other-hover --
var a.Other a.Thing
-- Thing-definition --
godef/a/d.go:5:6-11: defined here as Thing struct {
	Member string //@Member
}

-- Thing-definition-json --
{
	"span": {
		"uri": "file://godef/a/d.go",
		"start": {
			"line": 5,
			"column": 6,
			"offset": 30
		},
		"end": {
			"line": 5,
			"column": 11,
			"offset": 35
		}
	},
	"description": "Thing struct {\n\tMember string //@Member\n}"
}

-- Thing-hover --
Thing struct {
	Member string //@Member
}
-- Things-definition --
godef/a/d.go:11:6-12: defined here as func a.Things(val []string) []a.Thing

-- Things-definition-json --
{
	"span": {
		"uri": "file://godef/a/d.go",
		"start": {
			"line": 11,
			"column": 6,
			"offset": 113
		},
		"end": {
			"line": 11,
			"column": 12,
			"offset": 119
		}
	},
	"description": "func a.Things(val []string) []a.Thing"
}

-- Things-hover --
func a.Things(val []string) []a.Thing
